src/components/Card.tsx

```diff
@@ -0,0 +1,333 @@
+import * as React from 'react';
+import { Post, getAuthorImageUri } from '../models/Post';
+import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
+import { Colors, DefaultStyle } from '../styles';
+import { View, ActivityIndicator, TouchableOpacity, TouchableWithoutFeedback, Dimensions, Platform, StyleSheet, Image, Text, Linking } from 'react-native';
+import { TouchableView } from './TouchableView';
+import { DateUtils } from '../DateUtils';
+import { Utils } from '../Utils';
+import { ImageView } from './ImageView';
+import { isSwarmLink } from '../Swarm';
+import { ImageData } from '../models/ImageData';
+import Markdown from 'react-native-easy-markdown';
+import { ErrorBoundary } from './ErrorBoundary';
+
+const WindowWidth = Dimensions.get('window').width;
+
+interface CardProps {
+    post: Post;
+    isSelected: boolean;
+    showSquareImages: boolean;
+    togglePostSelection: (post: Post) => void;
+    openPost: (post: Post) => void;
+    onDeleteConfirmation: (post: Post) => void;
+    onSharePost: (post: Post) => void;
+    navigate: (view: string, {}) => void;
+}
+
+export const Card = (props: CardProps) => {
+    if (props.post.images.length === 0) {
+        return <CardWithText {...props}/>;
```

Comment:
This is not necessary, the else branch renders the same when there are no images and without CardWithText there is no code duplication.

Reply:
There are some minor differences, but I just moved the code :-)